src/Analyzers/Core/Analyzers/SimplifyLinqExpression/AbstractSimplifyLinqExpressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,144 @@
+            {
+                if (typeof(Enumerable)?.FullName is string fullyQualifiedName)
+                {
+                    enumerableType = compilation.GetTypeByMetadataName(fullyQualifiedName);
+                    return enumerableType is not null;
+                }
+
+                enumerableType = null;
+                return false;
+            }
+
+            static bool TryGetLinqWhereExtensionMethods(INamedTypeSymbol enumerableType, out ImmutableArray<IMethodSymbol> whereMethods)
+            {
+                whereMethods = enumerableType.GetMembers(nameof(Enumerable.Where)).OfType<IMethodSymbol>().ToImmutableArray();
+                return !whereMethods.IsEmpty;
+            }
+
+            static bool TryGetLinqMethodsThatDotNotReturnEnumerables(INamedTypeSymbol enumerableType, out ImmutableArray<IMethodSymbol> linqMethods)
+            {
+                using var _ = ArrayBuilder<IMethodSymbol>.GetInstance(out var linqMethodSymbolsBuilder);
+                foreach (var methodName in _nonEnumerableReturningLinqMethodNames)
+                {
+                    linqMethodSymbolsBuilder.AddRange(enumerableType.GetMembers(methodName).OfType<IMethodSymbol>());
+                }
+
+                if (linqMethodSymbolsBuilder.Count == 0)
+                {
+                    linqMethods = default;
+                    return false;
+                }
+
+                linqMethods = linqMethodSymbolsBuilder.ToImmutable() ;
+                return true;
+            }
+        }
+
+        public void AnalyzeInvocationOperation(OperationAnalysisContext context, ImmutableArray<IMethodSymbol> whereMethods, ImmutableArray<IMethodSymbol> linqMethods)
+        {
+            if (context.Operation.Syntax.GetDiagnostics().Any(diagnostic => diagnostic.Severity == DiagnosticSeverity.Error))
+            {
+                // Do not analyze linq methods that contain diagnostics.
+                return;
+            }
+
+            if (context.Operation is not IInvocationOperation invocation ||
+                invocation.Arguments.Length != 1)
+            {
+                return;
+            }
+
+            if (invocation.Arguments.Single().Children.FirstOrDefault() is not IInvocationOperation previousInvocationInChain)
+            {
+                // Invocation is not part of a chain of invocations (i.e. Where(x => x is not null).First())
+                return;
+            }
+
+            // Verify this is 'Where' followed by a non-enumerable returning method
+            if (IsInvocationNonEnumerableReturningLinqMethod(invocation) &&
+                IsWhereLinqMethod(previousInvocationInChain) &&
+                previousInvocationInChain.Arguments.Length == 2)
+            {
+                var memberAccessExpressionLocation = previousInvocationInChain.Syntax.GetLocation();
+                var lambdaExpressionLocation = previousInvocationInChain.Arguments.Last().Syntax.GetLocation();
```

Comment:
we need to make sure these are correctly typed.  remember that hte compiler can rewwrite anything into IOps (including things like linq queries), so there's no guarantee this syntax i hte form you expect.